### DISCUSSION TITLE
openmdx build from source fails

### DISCUSSION THREAD

**Discussion by nyemiks** [openmdx/openmdx#46] [Q&A] [answered]
I am trying to set up the latest openmdx 2.17.9. 

gradle assemble fails

below is the log

> Task :core:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':core:compileJava'.
> invalid flag: --release

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 28s
5 actionable tasks: 5 executed

C:\openmdx_sdk\openmdx_2179\openmdx>

**Comment by wfro64** [comment, marked as answer]
The prerequisites as described [here](https://github.com/openmdx/openmdx-documentation/blob/master/Sdk/BuildFromSource.md) must be met for a successful build. It is assumed that a JDK 11 will be used. The release target `VERSION_1_8` is set in the `compileJava` task in `build.gradle.kts`:

`options.release.set(Integer.valueOf(targetPlatform.getMajorVersion()))`

JDK 8 can also be used for the build. In this case you have to disable this option since it is not known to the JDK 8 compiler. You have to do this for all sub-projects: `core`, `client`, `portal`, `security`, `test-core`, `test-security`, `tomcat`.

**Reply by nyemiks** [reply]
I have been able to build from source with java 8. Thanks !

**Comment by javac9** [comment]
I also have an issue when I run `gradle assemble` (version 2.17.10).

I tried Java 11:
`java -version`: 
`
openjdk version "11.0.12" 2021-07-20
OpenJDK Runtime Environment Temurin-11.0.12+7 (build 11.0.12+7)
OpenJDK 64-Bit Server VM Temurin-11.0.12+7 (build 11.0.12+7, mixed mode)
`

`echo %JAVA_HOME%`: `C:\JDK\OpenJDK-11`

```
> Task :security:compileJava
Note: C:\OpenMDX\openmdx\security\src\main\java\org\openmdx\resource\ldap\ldif\Repository.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :client:compileJava FAILED
C:\OpenMDX\openmdx\client\src\main\java\org\ietf\jgss\Oid.java:10: error: package sun.security.util does not exist
import sun.security.util.DerValue;
                        ^
C:\OpenMDX\openmdx\client\src\main\java\org\ietf\jgss\Oid.java:11: error: package sun.security.util does not exist
import sun.security.util.DerOutputStream;
                        ^
C:\OpenMDX\openmdx\client\src\main\java\org\ietf\jgss\Oid.java:12: error: package sun.security.util does not exist
import sun.security.util.ObjectIdentifier;
                        ^
C:\OpenMDX\openmdx\client\src\main\java\org\ietf\jgss\Oid.java:33: error: cannot find symbol
    private ObjectIdentifier oid;
            ^
  symbol:   class ObjectIdentifier
  location: class Oid
C:\OpenMDX\openmdx\client\src\main\java\org\ietf\jgss\Oid.java:48: error: cannot find symbol
            oid = new ObjectIdentifier(strOid);
                      ^
  symbol:   class ObjectIdentifier
```

**Reply by nyemiks** [reply]
The only thing that comes to my mind is that I think openmdx is qualified for jdk 8. So to use jdk 11 it's possible jdk 11 has removed some packages (jar) that were in jdk 8. You might have to research this missing jars and add them.

I am sure someone from openmdx will know what is causing this problem.

**Comment by DlRTY-HARRY** [comment]
openMDX will be updated to solve the issues mentioned above and to support different flavours:

- openMDX 2 & 3 will be targeted to Java 8/Jakarta 8 environments
- openMDX 4 & 5 will be targeted to Java 21/Jakarta 10 environments
